src/v/raft/prevote_stm.cc

```diff
@@ -64,32 +64,43 @@ prevote_stm::process_reply(vnode n, ss::future<result<vote_reply>> f) {
             voter_reply->second._is_failed = true;
             voter_reply->second._is_pending = false;
         } else {
-            auto r = f.get0();
-            if (r.has_value()) {
-                auto v = r.value();
-                if (v.log_ok) {
-                    vlog(
-                      _ctxlog.trace,
-                      "prevote ack: node {} caught up with a candidate",
-                      n);
-                    voter_reply->second._is_ok = true;
-                    voter_reply->second._is_pending = false;
+            if (f.failed()) {
+                // This can be e.g. a semaphore_timed_out
+                vlog(
+                  _ctxlog.trace,
+                  "prevote ack: node {} threw exception {}",
+                  f.get_exception(),
+                  n);
+                voter_reply->second._is_failed = true;
+                voter_reply->second._is_pending = false;
+            } else {
+                auto r = f.get0();
```

Comment:
> raft: fix abandoned exceptional future in prevote_stm

@jcsp I can't see an issue in the previous version of this code related to abandoned exceptional future. if `f` were exceptional, then `f.get0()` would have thrown and been caught below. Following `future::get0` the exception has been consumed out of the future so the future when destroyed will no longer complain about being an abandoned exceptional future.